rfcs/040-tei_adapter/README.md

```diff
@@ -68,43 +68,117 @@ The biggest disadvantage is that we don't have control or visibility if the Webh
 
 #### Pull the changes from GitHub at regular intervals:
   
-This means having a Lambda that regularly requests changes from GitHub within subsequent time intervals. 
-This is an approach we already use in the Sierra adapter and in the Calm adapter.
+This means regularly request changes from GitHub within certain time intervals. 
 The big advantage of this approach is that we control what happens if one request fail, how we should be notified
 and what action we should take.
 
 The proposal in this RFC is to use a pull approach so that we can be confident that all updates have either gone through
 successfully, or we are notified of any failure, and we can retry.
 
-This is a diagram of the proposed architecture: ![](architecture.png)
+#### How to query GitHub
 
-The Lambda in the picture should generate windows and send them to the TEI adapter
+We can ask the GitHub API for commits that happened in a certain time window with a query like: 
+``` 
+GET https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/commits?since=2021-01-20T10:00:00&until=2021-03-22T18:01:00&branch=master
+```
+This gives us all the commits that have happened in the time window passed. 
+The problem with this approach is that if there is a feature branch merged into master, the commits 
+coming from the merge will be shown at the time they were created in the feature branch. This means that a query for 
+commits within a window can return different results depending on whether the query is made before or after a merge. 
+To not loose changes coming from a merge, we would need to chase the commits coming from that branch when we see a merge commit, 
+which in turn means keeping track of the commits we have already seen and effectively keeping track of the
+history of the repo, which is very fiddly and error prone.
+
+Another approach is to ask the GitHub API for trees. Trees are a representation of the file hierarchy in the repo at a 
+certain commit. The following request gives us all the files in the repository at the latest commit in master:
+```
+https://api.github.com/repos/wellcomecollection/wellcome-collection-tei/git/trees/master?recursive=true
+```
 
+This is a small sample of the response:
+```yaml
```

Comment:
```suggestion
```json
```

Reply:
ah yes, I know. I did this on purpose because I prefer how the json is rendered when markdown thinks is yaml 😬